src/models/PrimaryAuth.js

```diff
@@ -106,12 +106,21 @@ function (Okta, BaseLoginModel, CookieUtil, Enums) {
 
       var primaryAuthPromise;
 
-      if (this.appState.get('isUnauthenticated') && !this.settings.get('features.passwordlessAuth')) {
-        primaryAuthPromise = this.doTransaction(function (transaction) {
-          var authClient = this.appState.settings.authClient;
-          return this.doPrimaryAuth(authClient, signInArgs, transaction.authenticate);
-        });
-      } else {
+      var authClient = this.appState.settings.authClient;
+
+      if (this.appState.get('isUnauthenticated')) {
+        // bootstrapped with stateToken
+        if (!this.settings.get('features.passwordlessAuth')) {
+          primaryAuthPromise = this.doTransaction(function (transaction) {
+            var authClient = this.appState.settings.authClient;
+            return this.doPrimaryAuth(authClient, signInArgs, transaction.authenticate);
```

Comment:
out of curiosity, why does this flow return immediately and not register all the fail and finish callbacks?

Reply:
Its similar to other flows, where we return a promise which  `primaryAuthPromise` is set to and the  
success,fail callbacks are handled https://github.com/okta/okta-signin-widget/blob/98fbdc84e4898a71a70f9866a9ace972dc83fc0d/src/models/PrimaryAuth.js#L130

Reply:
Does the new pipeline supports primaryAuth without passwordless? If `isIdxStateToken` is true here, this will break since we need to call `translation.login` instead.